internal/controller/nab_velero_backupspec.go

```diff
@@ -0,0 +1,136 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"context"
+	"errors"
+
+	"github.com/go-logr/logr"
+	velerov1api "github.com/vmware-tanzu/velero/pkg/apis/velero/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	nacv1alpha1 "github.com/migtools/oadp-non-admin/api/v1alpha1"
+	"github.com/migtools/oadp-non-admin/internal/common/constant"
+	"github.com/migtools/oadp-non-admin/internal/common/function"
+)
+
+// CreateVeleroBackupSpec creates or updates a Velero Backup object based on the provided NonAdminBackup object.
+//
+// Parameters:
+//
+//	ctx: Context for the request.
+//	log: Logger instance for logging messages.
+//	nab: Pointer to the NonAdminBackup object.
+//
+// The function generates a name for the Velero Backup object based on the provided namespace and name.
+// It then checks if a Velero Backup object with that name already exists. If it does not exist, it creates a new one.
+// The function returns boolean values indicating whether the reconciliation loop should exit or requeue
+func (r *NonAdminBackupReconciler) CreateVeleroBackupSpec(ctx context.Context, log logr.Logger, nab *nacv1alpha1.NonAdminBackup) (exitReconcile bool, requeueReconcile bool, errorReconcile error) {
+	logger := log.WithValues("NonAdminBackup", nab.Namespace)
+
+	veleroBackupName := function.GenerateVeleroBackupName(nab.Namespace, nab.Name)
+
+	if veleroBackupName == constant.EmptyString {
+		return true, false, errors.New("unable to generate Velero Backup name")
+	}
+
+	veleroBackup := velerov1api.Backup{}
+	err := r.Get(ctx, client.ObjectKey{Namespace: constant.OadpNamespace, Name: veleroBackupName}, &veleroBackup)
+
+	if err != nil && apierrors.IsNotFound(err) {
+		// Create VeleroBackup
+		// Don't update phase nor conditions yet.
+		// Those will be updated when then Reconcile loop is triggered by the VeleroBackup object
+		logger.Info("No backup found", nameField, veleroBackupName)
+
+		// We don't validate error here.
+		// This was already validated in the ValidateVeleroBackupSpec
+		backupSpec, errBackup := function.GetBackupSpecFromNonAdminBackup(nab)
+
+		if errBackup != nil {
+			// Should never happen as it was already checked
+			return true, false, errBackup
+		}
+
+		veleroBackup = velerov1api.Backup{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      veleroBackupName,
+				Namespace: constant.OadpNamespace,
+			},
+			Spec: *backupSpec,
+		}
+	} else if err != nil && !apierrors.IsNotFound(err) {
+		logger.Error(err, "Unable to fetch VeleroBackup")
+		return true, false, err
+	} else {
+		// We should not update already created VeleroBackup object.
```

Comment:
here, it should not generate a new name and try again :question:

Reply:
no no, I meant: is not this triggered when the generated velero backup name already exists :question: which can be considered corner case...

Reply:
I think that case is already covered in the line 58 conditional statement, right ?

Reply:
Fixed. Mistake during move of the code logic. This is corner case anyway.

I am wondering why linting didn't catch this as it's same logic in two if/else places.